OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/ide/instances/lcp_on_fields/problem/LCPOnFieldsEdgeFunctions.scala

```diff
@@ -0,0 +1,510 @@
+        }
+
+    override def composeWith(secondEdgeFunction: EdgeFunction[LCPOnFieldsValue]): EdgeFunction[LCPOnFieldsValue] =
+        secondEdgeFunction match {
+            case ObjectEdgeFunction(values2) =>
+                ObjectEdgeFunction((values -- values2.keys) ++ values2)
+
+            case PutFieldEdgeFunction(fieldName, value) =>
+                ObjectEdgeFunction((values - fieldName) + (fieldName -> value))
+
+            case IdentityEdgeFunction()   => this
+            case AllTopEdgeFunction(_)    => secondEdgeFunction
+            case AllBottomEdgeFunction(_) => secondEdgeFunction
+
+            case _ =>
+                throw new UnsupportedOperationException(s"Composing $this with $secondEdgeFunction is not implemented!")
+        }
+
+    override def meetWith(otherEdgeFunction: EdgeFunction[LCPOnFieldsValue]): EdgeFunction[LCPOnFieldsValue] =
+        otherEdgeFunction match {
+            case ObjectEdgeFunction(values2) =>
+                ObjectEdgeFunction(
+                    values.keySet
+                        .union(values2.keySet)
+                        .map { fieldName =>
+                            fieldName -> LinearConstantPropagationLattice.meet(
+                                values.getOrElse(fieldName, linear_constant_propagation.problem.VariableValue),
+                                values2.getOrElse(fieldName, linear_constant_propagation.problem.VariableValue)
+                            )
+                        }
+                        .toMap
+                )
+
+            case PutFieldEdgeFunction(fieldName, value) =>
+                ObjectEdgeFunction(
+                    (values - fieldName).map { case (fieldName2, _) =>
+                        fieldName2 -> linear_constant_propagation.problem.VariableValue
+                    } +
+                        (fieldName -> LinearConstantPropagationLattice.meet(
+                            value,
+                            values.getOrElse(fieldName, linear_constant_propagation.problem.VariableValue)
+                        ))
+                )
+
+            case IdentityEdgeFunction()   => this
+            case AllTopEdgeFunction(_)    => this
+            case AllBottomEdgeFunction(_) => otherEdgeFunction
+
+            case _ =>
+                throw new UnsupportedOperationException(s"Meeting $this with $otherEdgeFunction is not implemented!")
+        }
+
+    override def equalTo(otherEdgeFunction: EdgeFunction[LCPOnFieldsValue]): Boolean =
+        (otherEdgeFunction eq this) ||
+            (otherEdgeFunction match {
+                case ObjectEdgeFunction(values2) => values == values2
+                case _                           => false
+            })
+}
+
+/**
+ * Edge function for initializing an object.
+ *
+ * @author Robin Körkemeier
+ */
+object NewObjectEdgeFunction extends ObjectEdgeFunction(immutable.Map.empty) {
+    override def toString: String = "NewObjectEdgeFunction()"
```

Comment:
You can probably get rid of toString methods like this if you just make these objects `case object`s instead, which provide a default toString method